### ISSUE TITLE
Updates available notification shows when I pushed the change

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode-pull-request-github#3236]
PR: https://github.com/microsoft/vscode/pull/140196/

I pushed a commit via the terminal and the notification showed briefly afterwards, despite being the one that pushed it.

![image](https://user-images.githubusercontent.com/2193314/148566300-ab5a90a3-3488-436d-b3d6-7b377191494c.png)

ext: 0.35.202201069

**Comment by alexr00** [comment]
@Tyriar this doesn't happen every time, correct?

It looks like this can happen when you get unlucky and the poll-for-changes occurs right after you push but before we've finished updating all our state.

**Comment by Tyriar** [comment]
Yes I don't think it happens every time. I'm guessing checking the commit hash when there's changes to verify should be a lightweight action.

**Comment by alexr00** [comment]
Yep, that's what we currently do. It's just a matter of when we update the commit hash.
